KIPs/kip-2.md

```diff
@@ -0,0 +1,252 @@
+
+## Simple Summary
+<!--"If you can't explain it simply, you don't understand it well enough." Provide a simplified and layman-accessible explanation of the KIP.-->
+This documentation defines the format of a keystore file which is used to securely store private keys of a Klaytn account.
+
+## Abstract
+<!--A short (~200 word) description of the technical issue being addressed.-->
+Since a Klaytn account can have one or more private keys, the format can provide a better way of storing multiple private keys securely and collectively instead of storing raw private keys.
+
+## Motivation
+<!--The motivation is critical for KIPs that want to change the Klaytn protocol. It should clearly explain why the existing protocol specification is inadequate to address the problem that the KIP solves. KIP submissions without sufficient motivation may be rejected outright.-->
+Klaytn improves usability by decoupling addresses and key pairs as well as providing various account key types such as multi-signature and role-based keys.
+Due to this change, [keystore v3](https://github.com/ethereum/wiki/wiki/Web3-Secret-Storage-Definition) is insufficient for Klaytn.
+Klaytn keystore format v4 is suitable for Klaytn by storing multiple private keys while it is based on keystore v3.
+
+## Specification
+<!--The technical specification should describe the syntax and semantics of any new feature. The specification should be detailed enough to allow competing, interoperable implementations for any of the current Klaytn platforms (klaytn). -->
+
+The specification of Klaytn keystore format v4 in JSON for a Klaytn account is shown below.
+This format can express various Klaytn account key types such as a single key, multiple keys, and role-based keys. 
+
+```
+<keystore> :=
+{
+  "version": 4,
+  "id": <a 128-bit UUID in hexstring>,
+  "address": <Klaytn address, a 160-bit hexstring>,
+  "keyring": <keyring>
+}
+
+<keyring> := [<key> (, <key>)*] | [<keys> (, <keys>)*]
+
+<keys> := [] | [<key> (, <key>)*]
+
+<key> :=
+{
+  "cipher": "aes-128-ctr",
+  "ciphertext": <a 256-bit hexstring>,
+  "cipherparams": {
+    "iv": <a 128-bit hexstring>
+  },
+  "kdf": "scrypt" | "pbkdf2",
+  "kdfparams": <kdfparams-scrypt> | <kdfparams-pbkdf2>,
+  "mac": <a 256-bit hexstring>
+}
+
+<kdfparams-scrypt> :=
+{
+  "dklen": <an integer>,
+  "salt": <a 256-bit hexstring>,
+  "n": <an integer>,
+  "r": <an integer>,
+  "p": <an integer>,
+}
+
+<kdfparams-pbkdf2> :=
+{
+  "dklen": <an integer>,
+  "salt": <a 256-bit hexstring>,
+  "prf": "hmac-sha256",
+  "c": <an integer>,
+}
+
+```
+
+## Examples
+Examples are as following:
+
+1. A single key
+    ```
+    {
+        "version": 4,
+        "id": "b6c6bd8f-e3ba-43c1-af4b-1b7dfe13e5f9",
+        "address": "0xa8677855999712d23692a4cad77bb9396a1174cd",
+        "keyring": [
+            {
+                "ciphertext": "5cbd3cf235d02718f1cfac84157786d8e36fcf651e15f0c9a4303a2338697126",
+                "cipherparams": {
```

Comment:
How about formatting the cipherparams field the same way?
Some are all on one line, some are newline.

Reply:
Okay. I will use newlines. :)